cmd/ci-secret-generator/main.go

```diff
@@ -132,6 +134,33 @@ func (o *options) validateConfig() error {
 	return nil
 }
 
+func validateCommandOutput(cmdOutput []byte, validationCmd string) error {
+	if validationCmd == "" {
+		return nil
+	}
+	cmd := exec.Command("bash", "-o", "errexit", "-o", "nounset", "-o", "pipefail", "-c", validationCmd)
+	var outBuf, errBuf bytes.Buffer
+	cmd.Stdin = bytes.NewReader(cmdOutput)
+	cmd.Stdout = &outBuf
+	cmd.Stderr = &errBuf
+	if err := cmd.Run(); err != nil {
+		stderr := errBuf.Bytes()
+		stdout := outBuf.Bytes()
+		_, partialStreams := err.(*exec.ExitError)
+		return fmtExecCmdErr(execValidationCmdRunErrAction, validationCmd, err, stdout, stderr, !partialStreams)
+	}
+
+	stderr := errBuf.Bytes()
+	stdout := outBuf.Bytes()
+
+	if len(stderr) != 0 {
```

Comment:
@hongkailiu I'm not sure about this. Should we check the `stderr` other than the exit code?

Reply:
Generation and validation are fundamentally different, so I don't see why we should treat them the same.

This should just look at the exit code and print the combined (`stdout`+`stderr`) output on failure — but see next comments.

Reply:
It would be great if we could simplify it into something like:
```golang
func validateCommandOutput(cmdOutput []byte, validationCmd string) error {
	if validationCmd == "" {
		return nil
	}
	cmd := exec.Command("bash", "-o", "errexit", "-o", "nounset", "-o", "pipefail", "-c", validationCmd)
	cmd.Stdin = bytes.NewReader(cmdOutput)
	if out, err := cmd.CombinedOutput(); err != nil {
		return fmt.Errorf("failed to run validation command %q: %w\ncombined output:\n%s",
			validationCmd, err, out)
	}
	return nil
}
```

Reply:
checking on `stdeer` is the simplest validation before implementing the real validation: validation fails if `stderr` is not empty.

I agree that we do not need to check `stderr` if we validate combination of `stdout` and `stderr`.

Reply:
We treat the error output of the generation command specially because of the problem we had where deprecation warnings were missed, leading us to find problems only after the deprecation had happened.

Is that likely to happen in validation commands?  Is that worth the additional complexity of this code and of every validation command which will have to carefully manage its `stderr`?  (I don't know the answer, since I don't know what types of programs will be used for validation and their output)